modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/query/builder/FilterQueries.scala

```diff
@@ -0,0 +1,223 @@
+    *
+    * @param org        the organization filter
+    * @param filter     the filter expression to be applied
+    * @param pagination the pagination values
+    * @param term       the optional full text search term
+    */
+  def list(org: OrgId, filter: Filter, pagination: Pagination, term: Option[String])(implicit Q: ConfiguredQualifier[Id]): F[QueryResults[Id]] =
+    list(filter and orgExpr(org), pagination, term)
+
+  /**
+    * Lists all ids in the system within the specified domain that match the given filter.
+    *
+    * @param dom        the domain filter
+    * @param filter     the filter expression to be applied
+    * @param pagination the pagination values
+    * @param term       the optional full text search term
+    */
+  def list(dom: DomainId, filter: Filter, pagination: Pagination, term: Option[String])(implicit Q: ConfiguredQualifier[Id]): F[QueryResults[Id]] =
+    list(
+      filter
+        and orgExpr(dom.orgId)
+        and domExpr(dom),
+      pagination,
+      term)
+
+  /**
+    * Lists all ids in the system within the specified domain and that have the specified schema name that match
+    * the given filter.
+    *
+    * @param dom        the domain filter
+    * @param schemaName the schema name filter
+    * @param filter     the filter expression to be applied
+    * @param pagination the pagination values
+    * @param term       the optional full text search term
+    */
+  def list(dom: DomainId, schemaName: String, filter: Filter, pagination: Pagination, term: Option[String])(implicit Q: ConfiguredQualifier[Id]): F[QueryResults[Id]] =
+    list(
+      filter
+        and orgExpr(dom.orgId)
+        and domExpr(dom)
+        and schemaNameExpr(schemaName),
+      pagination,
+      term)
+
+  /**
+    * Lists all ids in the system conformant to the specified schema that match the given filter.
+    *
+    * @param schema     the schema filter
+    * @param filter     the filter expression to be applied
+    * @param pagination the pagination values
+    * @param term       the optional full text search term
+    */
+  def list(schema: SchemaId, filter: Filter, pagination: Pagination, term: Option[String])(implicit Q: ConfiguredQualifier[Id]): F[QueryResults[Id]] =
+    list(
+      filter
+        and orgExpr(schema.domainId.orgId)
+        and domExpr(schema.domainId)
+        and schemaNameExpr(schema.name)
+        and schemaVerExpr(schema.version),
+      pagination,
+      term)
+
+  /**
+    * Lists all outgoing ids linked to the if identified by ''id'' that match the given filter.
+    *
+    * @param id         the selected id (this)
+    * @param filter     the filter to apply to outgoing ids
+    * @param pagination the pagination values
+    * @param term       the optional full text search term
+    */
+  def outgoing(id: String, filter: Filter, pagination: Pagination, term: Option[String] = None)(implicit Q: ConfiguredQualifier[Id]): F[QueryResults[Id]] = {
```

Comment:
using strings is quite unsafe in this scenario; make it generic with a qualifier